### PR TITLE
Add ansible210 testing

### PR DESCRIPTION
**Pull request by ssbarnea**
Instead of using devel branch which is now 2.11, switch back to pypi versions for 2.10 testing.

Related: https://github.com/ansible/ansible/issues/70705

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -15,10 +15,7 @@ description =
 deps =
   ansible28: ansible>=2.8,<2.9
   ansible29: ansible>=2.9,<2.10
-  # TODO(ssbarnea): Replace replace the two tarbals created by @abadger1999 once
-  # we official (pre-)releases are available on pypi.org
-  ansibledevel: ansible-base @ https://toshio.fedorapeople.org/ansible/acd/ansible-base/ansible-base-2.10.0.dev1.tar.gz#egg=ansible-base
-  ansibledevel: ansible @ https://toshio.fedorapeople.org/ansible/acd/ansible/ansible-2.10.0.tar.gz#egg=ansible
+  ansibledevel: ansible>=2.10.0a1,<2.11
```
Comment (webknjaz) [inline tox.ini:18]: I'd call it 
```suggestion
  ansible210: ansible>=2.10.0a1,<2.11
```
because it's not really devel.

Comment (ssbarnea) [reply tox.ini:18]: In that case lots of other changes would be required, so we add all the 210 jobs.

Comment (webknjaz) [reply tox.ini]: Yep. But it's not devel. devel of ansible-base is on github.

File: tox.ini [hunk 2]
```diff
@@ -15,10 +15,10 @@ description =
 deps =
   ansible28: ansible>=2.8,<2.9
   ansible29: ansible>=2.9,<2.10
+  ansible210: ansible>=2.10.0a1,<2.11
   # TODO(ssbarnea): Replace replace the two tarbals created by @abadger1999 once
   # we official (pre-)releases are available on pypi.org
-  ansibledevel: ansible-base @ https://toshio.fedorapeople.org/ansible/acd/ansible-base/ansible-base-2.10.0.dev1.tar.gz#egg=ansible-base
-  ansibledevel: ansible @ https://toshio.fedorapeople.org/ansible/acd/ansible/ansible-2.10.0.tar.gz#egg=ansible
+  ansibledevel: ansible @ git+https://github.com/ansible/ansible#egg=ansible
```
Comment (webknjaz) [inline tox.ini:21]: This is incorrect. What this is pointing at is base:
```suggestion
  ansibledevel: ansible-base @ git+https://github.com/ansible/ansible.git
```

Comment (webknjaz) [reply tox.ini:21]: I'm curious if @abadger has any plans to build nightlies for ACD...

Comment (abadger) [reply tox.ini]: I don't at the moment.

Comment (webknjaz) [reply tox.ini]: @abadger could you point to the ACD build script? maybe we could generate something for ourselves.

Comment (abadger) [reply tox.ini]: https://github.com/ansible-community/antsibull/blob/main/docs/build-ansible.rst#building-ansible

For 2.10, we use the build-single strategy (not build-multiple or build-collection). The instructions have one problem due to a bug in the way the script assigns default values to the deps-file. Leave out `--deps-file ansible-build-data/2.10/acd-2.10.0.deps` when you build-single and it should work.  The deps file will end up in the same directory as the tarball then.

Comment (webknjaz) [reply tox.ini]: > python -m pip install --user --upgrade --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple ansible

@abadger FYI this doesn't make much sense: this does not set preference for indexes. Using just `--extra-index-url https://test.pypi.org/simple/` would give the same result.

File: .github/workflows/tox.yml [hunk 3]
```diff
@@ -82,15 +82,17 @@ jobs:
     name: >-
       py${{ matrix.python-version }}@${{ matrix.os }}
     runs-on: ${{ matrix.os }}
+    continue-on-error: ${{ matrix.experimental }}
```
Comment (webknjaz) [inline .github/workflows/tox.yml:85]:
```suggestion
    continue-on-error: >
      ${{
        matrix.experimental
        && true
        || false
      }}
```

File: .github/workflows/tox.yml [hunk 4]
```diff
@@ -82,15 +82,17 @@ jobs:
...
-      # fail-fast: false
+      fail-fast: false
       # max-parallel: 5
       # The matrix testing goal is to cover the *most likely* environments
       # which are expected to be used by users in production. Avoid adding a
       # combination unless there are good reasons to test it, like having
       # proof that we failed to catch a bug by not running it. Using
       # distribution should be prefferred instead of custom builds.
       matrix:
+        experimental: [false]
```
Comment (webknjaz) [inline .github/workflows/tox.yml:95]: this shouldn't be necessary

File: .github/workflows/tox.yml [hunk 5]
```diff
@@ -169,11 +177,11 @@ jobs:
         python -m tox
       env:
         TOXENV: ansible29
-    - name: "Test with tox: ansibledevel"
+    - name: "Test with tox: ansible210"
```
Comment (webknjaz) [inline .github/workflows/tox.yml:180]: I'm curious if
```suggestion
    - name: Test with tox: ${{ env.TOXENV }}
```
will render the var from the outer scope of the step-specific one...

File: .github/workflows/tox.yml [hunk 6]
```diff
@@ -82,8 +82,14 @@ jobs:
...
     runs-on: ${{ matrix.os }}
+    continue-on-error: >-
+      ${{
+        matrix.experimental
+        && true
+        || false
+      }}
     strategy:
-      # fail-fast: false
+      fail-fast: false
```
Comment (webknjaz) [inline .github/workflows/tox.yml:92]: @ssbarnea could you explain why this change is needed?

Comment (ssbarnea) [reply .github/workflows/tox.yml]: Not sure if needed, it was in docs example around using experimental jobs. I removed it in last patch.
